ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoiningNode.java

```diff
@@ -40,11 +45,30 @@ public interface JoiningNode {
    */
   void setKeyFormat(KeyFormat format);
 
+  static ImmutableMap<String, String> getProperties(FormatInfo formatInfo) {
```

Comment:
Rename to be more informative: `getPropertiesWithoutSchemaId()` or similar. 

Also, I think it'd be cleaner to have this be a method on `KeyFormat`/`ValueFormat` (e.g., `copyWithoutSchemaId()`) which can then delegate to an analogous method for `FormatInfo`, rather than having `JoiningNode` be responsible for maintaining this. If someone updates either `KeyFormat` or `FormatInfo` to contain a new property in the future, and doesn't realize that the copy in this file needs to add the new property as well, then that's easy room for introducing bugs.